### DISCUSSION TITLE
refit_full=True

### DISCUSSION THREAD

**Discussion by yaniv92648** [autogluon/autogluon#5070] [Q&A]
Hi,

In the [TimeSeriesPredictor](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.html) constructor, I'm passing prediction_length=90 & freq='D' (to predict for 90 days).
In the [TimeSeriesPredictor.fit](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.fit.html) method, I'm passing num_val_windows=1 & refit_full=True.
So, TimeSeriesPredictor trains according to the training data of the historical data up to 90 days ago, and then validates the best model according to the validation set of the recent 90 days.
Because I'm passing refit_full=True, so I think that after finding the best model according to the validation set, TimeSeriesPredictor trains this best model over all of the recent data (including the 90 days that were used for the validation set earlier, to find the best model in the 1st place), and using this best model to predict  future data with.

Is the above correct?

I'm asking because as we can see in the attached [documentation](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.fit.html), in the refit_full parameter, it says "AutoGluon will attempt to re-train all models" and not "AutoGluon will attempt to re-train the best model".
![refit](https://github.com/user-attachments/assets/af62ced1-58f5-4c27-93d8-c3bc0b724ced)

Thanks!

**Comment by shchur** [comment]
Please see #5071
